autogen/oai/client.py

```diff
@@ -1,22 +1,29 @@
 from __future__ import annotations
 
-import sys
-from typing import Any, List, Optional, Dict, Callable, Tuple, Union
-import logging
 import inspect
+import logging
+import sys
 import uuid
-from flaml.automl.logger import logger_formatter
+from typing import Any, Callable, Dict, List, Optional, Protocol, Tuple, Union
 
+from flaml.automl.logger import logger_formatter
 from pydantic import BaseModel
-from typing import Protocol
 
 from autogen.cache import Cache
 from autogen.io.base import IOStream
-from autogen.oai.openai_utils import get_key, is_valid_api_key, OAI_PRICE1K
+from autogen.logger.logger_utils import get_current_ts
+from autogen.oai.openai_utils import OAI_PRICE1K, get_key, is_valid_api_key
+from autogen.runtime_logging import log_chat_completion, log_new_client, log_new_wrapper, logging_enabled
 from autogen.token_count_utils import count_token
 
-from autogen.runtime_logging import logging_enabled, log_chat_completion, log_new_client, log_new_wrapper
-from autogen.logger.logger_utils import get_current_ts
+try:
+    from autogen.img_utils import format_message_contents_with_images, is_vision_model
+except ImportError:
+    ERROR: Optional[ImportError] = ImportError(
+        "Please install with pyautogen[lmm] option if you want to use vision model."
+    )
+    format_message_contents_with_images = object
+    is_vision_model = object
```

Comment:
Given that pillow is the only dependency that means we need this check then I'd rather see this just be part of the default dependencies and remove this check.

Looks like replicate is only used in llava, so that's reasonable to keep behind the optional arg.

Reply:
@sonichi  Would adding pillow into the main dependencies too risky? 

Eventually, we might want to officially support multimodal soon in the future. As Adam mentioned, Guidance and many other packages are supporting multimodals already. Do you think we should add pillow to `setup.py` now?

@jackgerrits Yes, I agree, replicate is optional.

Reply:
@jackgerrits  Just talked with Chi. Let's not support Pillow yet, and we can raise an error when PIL is revoked. The changes has been made.